backend/alert/models.py

```diff
@@ -405,27 +409,75 @@ def is_active(self):
+        is_waiting_for_close property. Used for database filtering of registrations (you cannot
+        filter by a property value); unpack the filters with a single star
+        (NOT a double star like is_active_filter).
+        Example:
+            Registration.objects.filter(*Registration.is_waiting_for_close_filter())
+
+        As is the case with any Q filters, you must unpack these filters positionally before any
+        kwarg filters. If you are unfamiliar with Q queries, see the following docs:
+        https://docs.djangoproject.com/en/3.1/topics/db/queries/#complex-lookups-with-q-objects
+
+        All of these conditions are straightforward except for the last. The last condition
+        effectively guarantees the next registration in this registration's resubscribe chain
+        (if it exists) hasn't been deleted or cancelled. If the user has auto-resubscribe on and
+        then cancels the next registration, we don't want the previous registration to send
+        a close notification. Based on the way registrations are shown to the user in PCA,
+        they would expect cancelling or deleting the head of a resubscribe chain to halt all
+        notifications, close or otherwise, from that chain; this condition guarantees that.
+        Note that the only way for a registration 2 away from this one would be canceled is if
+        an alert was sent for the next registration, meaning a close notification for this
+        registration must have already been sent so close_notification_sent=True, or otherwise
+        the user cancelled and resubscribed multiple times, which would mean the next registration
+        would be cancelled. In all cases, is_waiting_for_close would correctly be False based
+        on these filters.
+        """
+        return (
+            Q(notification_sent=True),
+            Q(close_notification=True),
+            Q(deleted=False),
+            Q(cancelled=False),
+            Q(close_notification_sent=False),
+            (
+                Q(resubscribed_to__isnull=True)
+                | (
+                    Q(resubscribed_to__isnull=False)  # SQL short-circuit logic is not guaranteed
+                    & Q(resubscribed_to__cancelled=False)
+                    & Q(resubscribed_to__deleted=False)
+                )
+            ),
+        )
 
     @property
     def is_waiting_for_close(self):
         """
-        Returns True iff the registration would send a close notification
-        when the watched section changes to closed
+        True if the registration is waiting to send a close notification to the user
+        once the section closes.  False otherwise.
         """
-        for k, v in self.is_waiting_for_close_filter().items():
-            if getattr(self, k) != v:
-                return False
-        return True
+
+        # WARNING: you should be frugal with your usage of this property, since it hits the DB
+        # each time it is called.
```

Comment:
Let me know if there is a (reasonably simple) way to implement the `is_waiting_for_close` property without hitting the db; the issue is I really prefer to have one source of truth for this property (which currently is `is_waiting_for_close_filter`), while still being able to filter Registrations by this property. Since I am using Q filters now I'm not sure how I can check an instance with the filters without hitting the db; there is probably a way but I'm guessing it would be very complicated and involve some heavy reliance on the details of Django's implementation of the Q function...